tests/test_parse_load_args.py

```diff
@@ -0,0 +1,142 @@
+            self.assertEqual(parse(["-d", dist.name]).distribution, dist)
+
+        with self.assertRaises((argparse.ArgumentError, SystemExit)):
+            parse(["-d", "invalidDistro"])
+        with self.assertRaises((argparse.ArgumentError, SystemExit)):
+            parse(["-d"])
+
+        for dist in Distribution:
+            with cd_temp_dir():
+                config = Path(getcwd()) / Path("mkosi.default")
+                config.write_text(f"[Distribution]\nDistribution={dist}")
+                self.assertEqual(parse([]).distribution, dist)
+
+    def test_machine_id(self):
+        id = uuid.uuid4().hex
+        load_args = parse(["--machine-id", id])
+        self.assertEqual(load_args.machine_id, id)
+        self.assertTrue(load_args.machine_id_is_fixed)
+        with self.assertRaises(MkosiException):
+            parse(["--machine-id", "notValidKey"])
+        with self.assertRaises((argparse.ArgumentError, SystemExit)):
+            parse(["--machine-id"])
+
+        with cd_temp_dir():
+            config = Path(getcwd()) / Path("mkosi.default")
+            config.write_text(f"[Output]\nMachineID={id}")
+            load_args = parse([])
+            self.assertEqual(load_args.machine_id, id)
+            self.assertTrue(load_args.machine_id_is_fixed)
+
+        with cd_temp_dir():
+            config = Path(getcwd()) / Path("mkosi.default")
+            config.write_text("[Output]\nMachineID=")
+            with self.assertRaises(MkosiException):
+                parse([])
+
+    def test_hostname(self):
+        self.assertEqual(parse(["--hostname", "name"]).hostname, "name")
+        with self.assertRaises(SystemExit):
+            parse(["--hostname", "name", "additional_name"])
+        with self.assertRaises(SystemExit):
+            parse(["--hostname"])
+
+        with cd_temp_dir():
+            config = Path(getcwd()) / Path("mkosi.default")
+            config.write_text("[Output]\nHostname=name")
+            self.assertEqual(parse([]).hostname, "name")
+
+        with cd_temp_dir():
+            config = Path(getcwd()) / Path("mkosi.default")
+            config.write_text("[Output]\nHostname=")
+            config = Path(getcwd()) / Path("hostname.txt")
+            self.assertEqual(parse([]).hostname, "")
+
+    def test_centos_brtfs(self):
+        with cd_temp_dir():
+            config = (getcwd()) / Path("mkosi.default")
+            for dist in (Distribution.centos, Distribution.centos_epel):
+                for release in range (2, 9):
+                    with open(config, 'w') as file:
+                        file.write(f"[Distribution]\nDistribution={dist}\n")
+                        file.write(f"Release={release}\n\n")
+                        file.write("[Output]\nFormat=gpt_btrfs")
+                    with self.assertRaises(MkosiException):
+                        parse([])
+
+
+        with cd_temp_dir():
+            config = (getcwd()) / Path("mkosi.default")
```

Comment:
getcwd() can be removed here and elsewhere I think. All paths that don't start with a '/' are relative paths which are resolved relative to the current working directory.